src/main/java/com/uber/cadence/workflow/Saga.java

```diff
@@ -0,0 +1,186 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.workflow;
+
+import com.uber.cadence.client.BatchRequest;
```

Comment:
Let's rename it to OperationsBatch (or you have better idea?).
And move to common package as client package doesn't make sense here.

Reply:
I decided to not implement the interface, just add the methods and call them addCompensate explicitly. Let me know what you think.